tool.go

```diff
@@ -555,7 +571,7 @@ func (fs serveCommandFileSystem) Open(name string) (http.File, error) {
 	if isPkg || isMap || isIndex {
 		// If we're going to be serving our special files, make sure there's a Go command in this folder.
 		s := gbuild.NewSession(fs.options)
-		pkg, err := gbuild.Import(path.Dir(name[1:]), 0, s.InstallSuffix(), fs.options.BuildTags)
+		pkg, err := gbuild.Import(path.Dir(strings.Trim(name, "/")), 0, s.InstallSuffix(), fs.options.BuildTags)
```

Comment:
> With this change the first character is sometimes a slash, but not always.

I see. I asked I'd like to understand exactly in what situations it'a slash, and in what situations it's not.

> `strings.Trim` seemed like a good way to conditionally drop the first character if it's a '/'. I'm new(ish) to go so I'd love to learn if there's a more idiomatic approach. :)

One immediate improvement suggestion would be to use [`strings.TrimLeft`](https://godoc.org/strings#TrimLeft) rather than `strings.Trim`, since you said you only want to drop the first character being `/`, not the last character.

However, I understand what's happening better now, and I have a suggestion for how you can go back to just having an unconditional trim of first character (which should be simpler to reason about).

The signature of this method used to be `Open(name string)`, and `name[1:]` used to be done to drop the first "/" that's a part of each `*http.Request.URL.Path`. Since you've re-arranged the code to be `Open(requestName string)`, I think you can move the the `[:1]` operation to happen earlier:

```Go
func (fs serveCommandFileSystem) Open(requestName string) (http.File, error) {
	name := path.Join(fs.serveRoot, requestName[1:])
```

Then this line can become simply:

```Go
pkg, err := gbuild.Import(path.Dir(name), 0, s.InstallSuffix(), fs.options.BuildTags)
```

What do you think of that?